streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceivedBlockTracker.scala

```diff
@@ -106,6 +106,12 @@ private[streaming] class ReceivedBlockTracker(
       timeToAllocatedBlocks(batchTime) = allocatedBlocks
       lastAllocatedBatchTime = batchTime
       allocatedBlocks
+    } else if (batchTime == lastAllocatedBatchTime) {
```

Comment:
Could you improve the unit test to actually verify the behavior. The unit test line you removed can be modified to verify that calling `batchTime` = `lastAllocatedBatchTime` is a no-op and does no allocation.